### PR TITLE
Add missing svgnet/data/ module and pointops PyTorch 2.x patch

### LINKED ISSUE #9: Missing data loader module (svgnet/data/) and pointops compatibility issue
## Issue

The released code is missing the `svgnet/data/` module (data loader for ArchCAD format). The README references `from svgnet.data import build_dataloader, build_dataset` but this module is not included in the repository.

Additionally, the `pointops` submodule (YodaEmbedding/pointops) has compatibility issues:
1. THC headers removed in PyTorch 2.x require manual patching
2. The `interpolation` function produces out-of-bounds indices when used with the DPSS decoder's `get_subscene_features`

## Steps to Reproduce
1. Clone repo
2. Install deps per README
3. Build pointops (requires THC header patches for PyTorch 2.4)
4. Create custom data loader matching the 9-field batch format
5. Training crashes at `pointops.interpolation` with CUDA index out of bounds

## Request
Could you please:
1. Release the `svgnet/data/` module (ArchCAD data loader)
2. Provide a Docker image or the exact compiled pointops `.so` you used
3. Or release pre-trained DPSS weights

Thank you for the great paper and dataset!

### PR DESCRIPTION
**Pull request by mohanssgit**
### 1. `svgnet/data/svg.py` — Data loader for ArchCAD-400k

The `svgnet/data/` module referenced in `tools/train.py` is missing from the repo. This adds a working implementation.

**What it does:**
- Loads JSON (primitives) + PNG (rasterized tiles) from the HuggingFace dataset
- Computes 10D features per primitive: angle, length, RGB, type one-hot, stroke width
- Produces the 9-field batch format expected by `SVGNet.forward()`:
  `(coords, feats, semantic_labels, offsets, lengths, layerIds, imgs, centers, json_files)`
- Auto-splits 70/10/20 (train/val/test) when no split file is provided
- Handles variable-length batching, coordinate normalization, and padding to minimum 64 points

### 2. `svgnet/data/__init__.py`

Exports `SVGDataset`, `build_dataset`, `build_dataloader` — matching the imports in `tools/train.py`.

### 3. `patch_pointops.sh` — PyTorch 2.x compatibility

The pointops CUDA extension includes `THC/THC.h`, which was removed in PyTorch 2.0. This script comments out those includes so it compiles on modern PyTorch.

```bash
bash patch_pointops.sh
cd modules/pointops && python setup.py install
```

---

## Known issue: pointops submodule

The `.gitmodules` references `YodaEmbedding/pointops`, but the codebase uses `pointops.queryandgroup()` and `pointops.sectorized_fps()` which **don't exist** in that fork.

These functions are available in [SymPointV2's pointops](https://github.com/nicehuster/SymPointV2/tree/master/modules/pointops). Replacing `modules/pointops/` with SymPointV2's version + applying the THC patch resolves this.

---

## Tested on

- **GPU:** A100 80GB
- **PyTorch:** 2.2.0 + CUDA 12.1
- **Dataset:** 41,097 samples → 28,767 train / 4,110 val / 8,220 test
- **Model:** loads successfully (100.8M parameters), all imports resolve

### CODE REVIEW

**Comment by mohanssgit** [comment]
@liuzhengjie08 Your review would be amazing here!
